CriticalMass/RequestManager.swift

```diff
@@ -15,46 +15,69 @@ import os.log
 
 @available(macOS 10.12, *)
 public class RequestManager {
-    private struct SendLocationPostBody: Codable {
-        var device: String
-        var location: Location
-    }
-
     private struct SendMessagePostBody: Codable {
         var device: String
         var messages: [SendChatMessage]
     }
 
-    private var hasActiveRequest = false
-
     private var dataStore: DataStore
     private var locationProvider: LocationProvider
     private var networkLayer: NetworkLayer
     private var idProvider: IDProvider
+    private var networkObserver: NetworkObserver?
+
+    private let operationQueue: OperationQueue = {
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 1
+
+        return queue
+    }()
 
     private var log = OSLog(subsystem: Bundle.main.bundleIdentifier ?? "", category: "RequestManager")
 
-    public init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 12.0, idProvider: IDProvider) {
+    public init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 12.0, idProvider: IDProvider, networkObserver: NetworkObserver?) {
```

Comment:
so 12 remains the default?

Reply:
I left 12 seconds as it was before refactoring, can you please explain more about 1 second interval? I mean I understand that frequent requests might be a problem for server, but how user can schedule requests every second, if we set it in code for 12 seconds? Do you thing we should increase intervals? Or there should be something like minimum waiting interval?

Reply:
sorry maybe i misunderstood. from the title of the pr i thought this should enable the user to dynamically set the poll interval.

Reply:
Actually no, we don't allow users to set poll interval. The main goal sounds like: if user lost internet connection and regain it later, we retrieve location and send update request as soon as possible. But next request will be send with poll interval of 12 seconds.

Reply:
:+1: